spec/reservation_tracker_spec.rb

```diff
@@ -0,0 +1,162 @@
+        @hotel.book_reservation(room, Date.new(2018,6,4), Date.new(2018,7,7))
+      end
+
+      expect(
+        @hotel.available_rooms(Date.new(2018,6,4), Date.new(2018,7,7))
+      ).must_be_instance_of(Array)
+      expect(
+        @hotel.available_rooms(Date.new(2018,7,6), Date.new(2018,7,7))
+      ).must_equal([*11..20])
+
+
+      [*11..19].each do |room|
+        @hotel.book_reservation(room, Date.new(2018,6,4), Date.new(2018,7,7))
+      end
+
+      expect(
+        @hotel.available_rooms(Date.new(2018,6,10), Date.new(2018,7,7))
+      ).must_equal([20])
+    end
+  end
+
+  describe 'book_reservation' do
+    before do
+      @hotel.book_reservation(1, Date.new(2018,7,4), Date.new(2018,7,8))
+      @hotel.book_reservation(1, Date.new(2018,8,4), Date.new(2018,8,5))
+      @hotel.book_reservation(2, Date.new(2018,6,4), Date.new(2018,7,7))
+    end
+
+    it 'raises error if date is not in correct format' do
+      expect{
+        @hotel.book_reservation(1, '20180503', Date.new(2018,5,9))
+      }.must_raise(ArgumentError)
+      expect{
+        @hotel.book_reservation(1, Date.new(2018,5,9), '20180503')
+      }.must_raise(ArgumentError)
+    end
+
+    it 'raises error if room is unavailable' do
+      expect{
+        @hotel.book_reservation(1, Date.new(2018,7,7), Date.new(2018,7,10))
+      }.must_raise(StandardError)
+      expect{
+        @hotel.book_reservation(2, Date.new(2018,5,1), Date.new(2018,6,4))
+      }.must_raise(StandardError)
+    end
+
+    it 'successful book returns reservation with matching dates' do
+      reservation = @hotel.book_reservation(5, Date.new(2018,7,7), Date.new(2018,7,10))
+
+      expect( reservation ).must_be_instance_of(Hotel::Reservation)
+      expect( reservation.start_date ).must_equal(Date.new(2018,7,7))
+      expect( reservation.end_date ).must_equal(Date.new(2018,7,10))
+    end
+
+    it 'successful book increases total reservations array by one' do
+      initial_reservations = @hotel.reservations.length
+      reservation = @hotel.book_reservation(5, Date.new(2018,7,7), Date.new(2018,7,10))
+
+      expect( @hotel.reservations.length - initial_reservations).must_equal(1)
+    end
+
+    it 'raise error if trying to book a reservation that is reserved in a block' do
+      @hotel.create_block([1,2], Date.new(2018,1,1), Date.new(2018,2,1))
+
+      expect{
+        @hotel.book_reservation(1, Date.new(2017,12,31), Date.new(2018,1,1))
+      }.must_raise(StandardError)
+    end
+  end
+
+  describe 'create_block' do
+    it 'returns block object' do
+      expect(
+        @hotel.create_block([*1..5], Date.new(2018,9,17), Date.new(2018,9,20))
+      ).must_be_instance_of(Hotel::RoomBlock)
+    end
+
+    it 'successful call adds a block instance to blocks attribute' do
+      initial_blocks = @hotel.blocks.length
+      @hotel.create_block([*1..5], Date.new(2018,9,17), Date.new(2018,9,20))
+
+      expect( @hotel.blocks.length - initial_blocks ).must_equal(1)
```

Comment:
I think you set up this test well -- It definitely comes across clearly that you're just looking for the change of the number of blocks to change. Bravo!